perf_events.go

```diff
@@ -47,6 +47,9 @@ type PerfEvents struct {
 	stopChannel    chan struct{}
 	wg             sync.WaitGroup
 
+	started      bool
+	startStopMtx sync.Mutex
```

Comment:
I think the mutex (and concurrency support) might be overkill - if I'm not mistaken the restructuring of the initialization into `startLoop` and out of `loop` would fix your original issue (and is much appreciated!)

How would you feel about making it just that gentler fix?

(My concern is that there are probably other things not-concurrency-safe in here, so going all the way to external-concurrency-safe for these is likely to distort client expectations.)